### PR TITLE
WIP: ZTF API

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR provides a submission interface for the ZTF interface.

### CODE REVIEW

**Comment by dmitryduev** [comment]
@mcoughlin this is really nice, but I would close it here and move to Fritz as an extension. @dannygoldstein do you agree? https://github.com/fritz-marshal/fritz/blob/master/extensions/skyportal/skyportal/handlers/api/kowalski_filter.py would be a good reference for talking to K from SP/F - we have `penquins` as a requirement there, so processing requests would be more robust.

**Comment by mcoughlin** [comment]
I definitely understand having the talking to K from SP/F bit in fritz, but I am a bit surprised everything else would be there too, given all the instruments are otherwise in SP.

**Comment by dmitryduev** [comment]
yea, that's right, but can you separate the concerns easily? @dannygoldstein @stefanv @kmshin1397 @acrellin what do you guys think?

File: requirements.txt [hunk 1]
```diff
@@ -39,3 +39,5 @@ twilio>=6.45.3
 jsonschema-to-openapi>=0.2.1
 responses>=0.12.0
 numba>=0.51.2
+sshtunnel
+pyjwt
```
Comment (acrellin) [inline requirements.txt:43]: `pyjwt` is already a requirement in `baselayer`

**Comment by acrellin** [comment]
The general followup request test improvements and the new instrument in the demo data should go in SkyPortal, but it looks like pretty much everything else is ZTF-specific, right?

**Comment by kmshin1397** [comment]
Hmm is there a reason not to have any ZTF references in particular at all in the SkyPortal code? It does feel like since LT and SEDM are already part of the codebase with code/URLs specific to those APIs that at least some form of the ZTF API should also be allowed in here. Perhaps we could stub out just the `authenticate` and `query_kowalski` parts here and fill those in on the Fritz side? The `query_kowalski` method could be renamed to something generic like `send_ztf_request` for example, and both that and `authenticate` could just return a `NotImplementedError` in SkyPortal?

**Pull request closed without merging**